### ISSUE TITLE
MODE 2 not working on //e

### ISSUE THREAD

**Issue by bobbimanners** [bobbimanners/Applecorn#235]
Somehow I broke it.  Works on GS still.

**Comment by bobbimanners** [comment]
It is crashing on 65816 code in `CLREOLGS`.  It shouldn't be getting there.  I think `VDUBANK` is the problem here.

**Comment by bobbimanners** [comment]
Seems this got broken a while back, early in the SHR work.  The last working build I could find doing a quick bisect was this one:

_commit 0395d560123c136fa89325347ff23cfcefcb01c8
Author: Bobbi Webber-Manners <bobbi.manners@gmail.com>
Date:   Fri Dec 23 21:11:32 2022 -0500

    Char plotting works in MODE 0. No scroll yet._

**Comment by bobbimanners** [comment]
Broken by end of day Dec 25.

```
commit 32b129c85907b2116e0d889e2e0e6dd66bd0a786
Author: Bobbi Webber-Manners <bobbi.manners@gmail.com>
Date:   Sun Dec 25 14:40:50 2022 -0500

    Set text mode 40/80 properly in SHR modes (for COPY)
```

**Comment by bobbimanners** [comment]
Broken by end of day Dec 24.

```
commit acedd38cc1828cb6834d99889a21b15f53c497c1
Author: Bobbi Webber-Manners <bobbi.manners@gmail.com>
Date:   Sat Dec 24 23:27:08 2022 -0500

    Use ZP for SHR char plotting - little bit faster
```

**Comment by bobbimanners** [comment]
This one works still:

```
commit e5eabbac9705c042ff1fa4b8eda6e1d9bb3ac1a9
Author: Bobbi Webber-Manners <bobbi.manners@gmail.com>
Date:   Fri Dec 23 23:26:16 2022 -0500

    SHR 640 mode - support for text colours

```

**Comment by bobbimanners** [comment]
Broken in this build:
```
commit e2f13be9f894f42e89e5c8be79198502bc6490ec
Author: Bobbi Webber-Manners <bobbi.manners@gmail.com>
Date:   Sat Dec 24 20:16:20 2022 -0500

    Scrolling works in 640 SHR mode now.

```

**Comment by bobbimanners** [comment]
Broken here too:
```
commit 0e746972713692adaf55b3280aac961c6b4ae266
Author: Bobbi Webber-Manners <bobbi.manners@gmail.com>
Date:   Sat Dec 24 01:06:40 2022 -0500

    Fix #223. Also CLREOL (sorta) works in SHR mode.
```

**Comment by bobbimanners** [comment]
OK, so managed to deduce it is `SHRCLREOL` that is getting invoked on the //e somehow in HGR mode and ...bang!
